### PR TITLE
fix : Resolve issues with empty partition key query not returning anything

### PR DESCRIPTION
**Pull request by garrettbaski**
Fixing issue #2078. The issue was caused by checking if the partition key existed by doing !!<object>.PartitionKey. In JS this resolves to false if the value is an empty string. Instead, do a "hasOwnProperty" on the partition key to check if the field exists or not instead of checking if the value exist to solve this problem.

### CODE REVIEW

File: src/table/persistence/QueryInterpreter/IQueryContext.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ export type IQueryContext = Entity | Table;
  * @returns {boolean}
  */
 export function isEntity(context: IQueryContext): context is Entity {
-  return !!(<Entity>context).PartitionKey;
+  return !!(<Entity>context).hasOwnProperty('PartitionKey');
```
Comment (notheotherben) [inline src/table/persistence/QueryInterpreter/IQueryContext.ts:12]: optional nit: I think the leading `!!` is redundant here as `.hasOwnProperty` is already a boolean value.

**Review by notheotherben — APPROVED** [review]
Looks good to me, thank you @garrettbaski for getting this sorted.

**Comment by XiaoningLiu** [comment]
Thanks @garrettbaski! Looks good, and great if able to further update per @notheotherben's suggestion.

**Comment by notheotherben** [comment]
(for what it's worth, my suggestion isn't a functional blocker, just a stylistic one)

File: src/table/persistence/QueryInterpreter/IQueryContext.ts (hunk 1 above)
Comment (garrettbaski) [reply src/table/persistence/QueryInterpreter/IQueryContext.ts]: I went ahead and removed the ```!!```

**Review by EmmaZhu — APPROVED** [review]

**Review by blueww — APPROVED** [review]

**Pull request merged**